scipy/_lib/_array_api_util.py

```diff
@@ -0,0 +1,65 @@
+"""Utility functions to use Python Array API compatible libraries.
+
+For the context about the Array API see:
+https://data-apis.org/array-api/latest/purpose_and_scope.html
+
+The SciPy use case of the Array API is described on the following page:
+https://data-apis.org/array-api/latest/use_cases.html#use-case-scipy
+"""
+
+import numpy as np
+
+from scipy._lib import _pep440
+
+_NUMPY_INCLUDES_ARRAY_API = _pep440.parse(np.__version__) >= _pep440.Version(
+    "1.22.0"
+)
+
+
+def _get_namespace(*arrays):
```

Comment:
Personally I am not a fan of `get` something. I would prefer something even more explicit like `namespace_from_array` or something with array API in the name. Cf other comment on `_concatenate` about being explicit.

Reply:
> `namespace_from_array`

+1. A better name.

Reply:
I'm fine with dropping the underscores, it's just what I noticed is used in SciPy, for example [`scipy._lib._util_asarray_validated`](https://github.com/scipy/scipy/blob/93939bb3dfeeb2a99ea23bba361fbf6df771155b/scipy/_lib/_util.py#L241).

I don't have a preference for the name of this function. Array API is mentioned in the module name `scipy._lib._array_api_util`. I can change it to `scipy._lib._array_api_util._namespace_from_arrays`.

Reply:
For the underscore we are a bit inconsistent yes. Let's see what others think.